mq/mq.go

```diff
@@ -35,6 +35,24 @@ type Client struct {
 
 type Option func(c *Client) error
 
+// handleChannelClose waits for notifications about closing amqpChan.
+// When amqpChan is closed we close conn.
+// On normal channel close err will be nil, on exception error will not be nil.
+// It is a blocking call.
+// TODO we should consider better error handling, e.g. recreate amqpChan instead of closing connection
+func handleChannelClose(amqpChan *amqp.Channel, conn *amqp.Connection) {
```

Comment:
What do you think of putting this into `ListenConnection` instead?

Reply:
I think if we make reconnection logic a part of this package we should run it automatically and as soon as we create channel or connection. So maybe in future instead of closing connection here we can add ListenConnection logic here

Reply:
Am I correct that`handleChannelClose` currently depends on `ListenConnectionAsync` being called? (referring to my comment in [another thread](https://github.com/trustwallet/go-libs/pull/168/files#r986734751), what will happen to the consumers if connection is closed but ListenConnectionAsync is not activated)
If that's so, shouldn't we group these reconnection logic into 1 function? I do not have any preference doing reconnection logic ASAP, or through another function call (like `ListenConnection`), but they need to be in 1 part (all or nothing).